cli/pkg/workspace/plugin/kubeconfig.go

```diff
@@ -132,187 +136,210 @@ func (o *UseWorkspaceOptions) Run(ctx context.Context) error {
 		return fmt.Errorf("current %q context not found", rawConfig.CurrentContext)
 	}
 
-	var newServerHost string
-	var workspaceType *tenancyv1alpha1.WorkspaceTypeReference
-	switch o.Name {
-	case "-":
-		prev, exists := o.startingConfig.Contexts[kcpPreviousWorkspaceContextKey]
-		if !exists {
-			return errors.New("no previous workspace found in kubeconfig")
-		}
-
-		newKubeConfig := o.startingConfig.DeepCopy()
-		if currentContext.Cluster == kcpCurrentWorkspaceContextKey {
-			oldCluster, found := o.startingConfig.Clusters[currentContext.Cluster]
-			if !found {
-				return fmt.Errorf("cluster %q not found in kubeconfig", currentContext.Cluster)
-			}
-			currentContext = currentContext.DeepCopy()
-			currentContext.Cluster = kcpPreviousWorkspaceContextKey
-			newKubeConfig.Clusters[kcpPreviousWorkspaceContextKey] = oldCluster
-		}
-		if prev.Cluster == kcpPreviousWorkspaceContextKey {
-			prevCluster, found := o.startingConfig.Clusters[prev.Cluster]
-			if !found {
-				return fmt.Errorf("cluster %q not found in kubeconfig", currentContext.Cluster)
-			}
-			prev = prev.DeepCopy()
-			prev.Cluster = kcpCurrentWorkspaceContextKey
-			newKubeConfig.Clusters[kcpCurrentWorkspaceContextKey] = prevCluster
-		}
-		newKubeConfig.Contexts[kcpCurrentWorkspaceContextKey] = prev
-		newKubeConfig.Contexts[kcpPreviousWorkspaceContextKey] = currentContext
-
-		newKubeConfig.CurrentContext = kcpCurrentWorkspaceContextKey
-
-		if err := o.modifyConfig(o.ClientConfig.ConfigAccess(), newKubeConfig); err != nil {
-			return err
-		}
-
-		newServerHost = newKubeConfig.Clusters[newKubeConfig.Contexts[kcpCurrentWorkspaceContextKey].Cluster].Server
-
-		bindings, err := o.getAPIBindings(ctx, o.kcpClusterClient, newServerHost)
-		if err != nil {
-			// display the error, but don't stop the current workspace from being reported.
-			fmt.Fprintf(o.ErrOut, "error checking APIBindings: %v\n", err)
-		}
-		if err = findUnresolvedPermissionClaims(o.Out, bindings); err != nil {
-			// display the error, but don't stop the current workspace from being reported.
-			fmt.Fprintf(o.ErrOut, "error checking APIBindings: %v\n", err)
-		}
-
-		return currentWorkspace(o.Out, newServerHost, shortWorkspaceOutput(o.ShortWorkspaceOutput), nil)
+	// LEGACY(mjudeikis): Remove once everybody gets used to this
+	if o.Name == "home" || strings.HasPrefix(o.Name, core.RootCluster.String()+":") {
```

Comment:
`o.Name == home`?

Reply:
Hmm, that case is below. Did we have special "home" logic?

Reply:
we did. I just kept it here. Maybe not needed anymore?